pkg/controller/services.go

```diff
@@ -325,15 +325,40 @@ func apicExtNetCons(conName string, tenantName string,
 	return apicapi.NewFvRsCons(enDn, conName)
 }
 
+//Helper function to check if a string item exists in a slice
+func stringInSlice(str string, list []string) bool {
+	for _, v := range list {
+		if v == str {
+			return true
+		}
+	}
+	return false
+}
+
+func validScope(scope string) bool {
+	validValues := []string{"", "context", "tenant", "global"}
+	if !stringInSlice(scope, validValues) {
+		return false
+	}
+	return true
+}
 func apicContract(conName string, tenantName string,
-	graphName string) apicapi.ApicObject {
+	graphName string, scopeName string) (apicapi.ApicObject, error) {
+	scopeName_norm := strings.ToLower(scopeName)
```

Comment:
nit: please stick to just camel case. scopeName_norm can be renamed as normScopeName.